### PR TITLE
[Merged by Bors] - chore(Topology/ContinuousFunction/Bounded): Rename instances

### PR DESCRIPTION
**Pull request by YaelDillies**
All the unnamed instances here were very long, and Moritz recently linked one of these to a newcomer.

Also slightly clean up the `Lattice` instance

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/ContinuousFunction/Bounded.lean [hunk 1]
```diff
@@ -1242,22 +1238,22 @@ theorem coe_mul (f g : α →ᵇ R) : ⇑(f * g) = f * g := rfl
...
       norm_ofNormedAddCommGroup_le _ (mul_nonneg (norm_nonneg _) (norm_nonneg _)) _
     -- Porting note: These 5 fields were missing. Add them.
     left_distrib, right_distrib, zero_mul, mul_zero, mul_assoc }
 
 end Seminormed
 
-instance nonUnitalNormedRing [NonUnitalNormedRing R] : NonUnitalNormedRing (α →ᵇ R) :=
-  { BoundedContinuousFunction.nonUnitalSeminormedRing,
-    BoundedContinuousFunction.normedAddCommGroup with }
+instance nonUnitalNormedRing [NonUnitalNormedRing R] : NonUnitalNormedRing (α →ᵇ R) where
```
Comment (Ruben-VandeVelde) [inline Mathlib/Topology/ContinuousFunction/Bounded.lean:1254]:
```suggestion
instance instNonUnitalNormedRing [NonUnitalNormedRing R] : NonUnitalNormedRing (α →ᵇ R) where
```

File: Mathlib/Topology/ContinuousFunction/Bounded.lean [hunk 2]
```diff
@@ -1516,58 +1508,52 @@ section NormedLatticeOrderedGroup
...
+        obtain ⟨C₁, hf⟩ := f.bounded
+        obtain ⟨C₂, hg⟩ := g.bounded
+        refine ⟨C₁ + C₂, fun x y ↦ ?_⟩
+        simp_rw [NormedAddCommGroup.dist_eq] at hf hg ⊢
+        exact (norm_inf_sub_inf_le_add_norm _ _ _ _).trans (add_le_add (hf _ _) (hg _ _)) }
+
+@[simp, norm_cast] lemma coe_sup (f g : α →ᵇ β) : ⇑(f ⊔ g) = ⇑f ⊔ ⇑g := rfl
+#align bounded_continuous_function.coe_fn_sup BoundedContinuousFunction.coe_sup
+
+@[simp, norm_cast] lemma coe_inf (f g : α →ᵇ β) : ⇑(f ⊓ g) = ⇑f ⊓ ⇑g := rfl
```
Comment (Ruben-VandeVelde) [inline Mathlib/Topology/ContinuousFunction/Bounded.lean:1539]: Let's add an alias for the `coeFn` → `coe` changes

File: Mathlib/Topology/ContinuousFunction/Bounded.lean [hunk 3]
```diff
@@ -1516,58 +1508,52 @@ section NormedLatticeOrderedGroup
...
+
+instance instSemilatticeSup : SemilatticeSup (α →ᵇ β) :=
+  DFunLike.coe_injective.semilatticeSup _ coe_sup
```
Comment (Ruben-VandeVelde) [inline Mathlib/Topology/ContinuousFunction/Bounded.lean:1542]: I don't know if these changes might have a performance impact. Maybe we could land the name changes and do the other changes separately.

**Review by Ruben-VandeVelde — CHANGES_REQUESTED** [review]

File: Mathlib/Topology/ContinuousFunction/Bounded.lean (hunk 3 above)
Comment (YaelDillies) [reply Mathlib/Topology/ContinuousFunction/Bounded.lean:1542]: This is very easy to answer

**Comment by YaelDillies** [comment]
!bench

**Comment by dupuisf** [comment]
Thanks!

bors r+

**Pull request closed without merging**